ewfs/experiment.py

```diff
@@ -0,0 +1,102 @@
+"""Run the EWFS experiment."""
+
+import itertools
+from collections import defaultdict
+import qiskit
+import qiskit_aer
+
+from ewfs.file_io import save_data
+from ewfs.scenario import EWFS
+from ewfs.facets import Facets
+
+
+DEFAULT_SETTINGS = [setting for setting in itertools.product(["peek", "reverse_1", "reverse_2"], repeat=2)]
+
+
+def run_experiment(
+    shots: int = 10_000,
+    num_trials: int = 2,
+    charlie_sizes: range = range(1, 3),
+    debbie_sizes: range = range(1, 2),
+    strategy: str = "majority_vote",
+    backend: qiskit.providers.Backend = qiskit_aer.Aer.get_backend("aer_simulator"),
+    settings: list[tuple[str, ...]] | None = None,
+    save: bool = False,
+    save_path: str | None = None,
+) -> dict:
+    """Run the EWFS experiment."""
+    settings = settings or DEFAULT_SETTINGS
+
+    # The tasks dictionary has a key that corresponds to the qubit
+    # system size with an associated value of the task for that size.
+    tasks: defaultdict = defaultdict(dict)
+
+    # Construct all circuits to be run over all qubit sizes.
+    friend_sizes = []
+    for charlie_size in charlie_sizes:
+        for debbie_size in debbie_sizes:
+            for trial in range(1, num_trials + 1):
+                circuits = {}
+
+                # Create circuits for each EWFS setting.  Note, we remove barriers in the Qiskit circuit.
```

Comment:
```suggestion
                # Create circuits for each EWFS setting.
```
Perhaps this comment is just out of date?